### ISSUE TITLE
Allow allowedExtensions to Support Any Custom File Format Without Predefined Restrictions

### ISSUE THREAD

**Issue by vicajilau** [miguelpruivo/flutter_file_picker#1658]
### Problem Statement

Currently, the `allowedExtensions` parameter in `flutter_file_picker` is subject to a predefined list of supported file formats. This restricts the ability of developers to handle completely custom file types that may not be included in the predefined list, such as proprietary formats (e.g., `.maso`).

This limitation forces developers to either:
- Work around the library by post-filtering the files after selection.
- Fork the library to support their specific needs.

---

### Proposed Solution

Make `allowedExtensions` fully customizable by allowing any string as an extension. Instead of validating against a predefined list, the package could directly pass the provided extensions to the underlying platform implementation.

---

### Expected Behavior

Developers should be able to provide any valid string as an extension in the `allowedExtensions` parameter, such as:

```dart
FilePickerResult? result = await FilePicker.platform.pickFiles(
  type: FileType.custom,
  allowedExtensions: ['maso', 'xyz', 'customformat'],
);
```
This should enable filtering for the specified custom file types without restrictions.

---

### Benefits
- Supports proprietary and less common file types.
- Enhances flexibility for diverse application scenarios.
- Reduces the need for workarounds or modifications to the library.

**Comment by MrCsabaToth** [comment]
I had problems before where I wanted to only import CSV and had to resort to let everything fly: https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/commit/6157af1ebe8b795b89bd9aec2fc78decfabbb6f5
Now I realize that I cannot pick any file whatsoever with that file picker either from my Google Drive or my local Download of my phone.

**Comment by vicajilau** [comment]
Yes @MrCsabaToth, I have applied the same solution. Letting the users can select any type of file and handle in Dart to search in the path if it is correct or not. 

```dart
if (!filePath.endsWith(".csv")) {
      throw FileInvalidException("The loaded file is not a CSV file.");
}
```
But in my opinion this is an avoidable boilerplate.

**Comment by MrCsabaToth** [comment]
> Yes @MrCsabaToth, I have applied the same solution. Letting the users can select any type of file and handle in Dart to search in the path if it is correct or not.
> 
> ```dart
> if (!filePath.endsWith(".csv")) {
>       throw FileInvalidException("The loaded file is not a CSV file.");
> }
> ```
> 
> But in my opinion this is an avoidable boilerplate. I don't see why there should be a list of supported file formats instead of supporting a custom format that can be dynamically configured by the developer.

My problem is that now no matter what I do I don't get back any file which was selected. Nada. Nuku. Zilch.

**Comment by doudar** [comment]
> > Yes [@MrCsabaToth](https://github.com/MrCsabaToth), I have applied the same solution. Letting the users can select any type of file and handle in Dart to search in the path if it is correct or not.
> > if (!filePath.endsWith(".csv")) {
> >       throw FileInvalidException("The loaded file is not a CSV file.");
> > }
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But in my opinion this is an avoidable boilerplate. I don't see why there should be a list of supported file formats instead of supporting a custom format that can be dynamically configured by the developer.
> 
> My problem is that now no matter what I do I don't get back any file which was selected. Nada. Nuku. Zilch.

Is this only on recent versions of Android? I'm having the same issue.

**Comment by MrCsabaToth** [comment]
> Is this only on recent versions of Android? I'm having the same issue. 

This was with Android 14. However in my case it turned out that a section in my pub spec was preventing generation to run and more things to execute, see more in https://github.com/TrackMyIndoorWorkout/TrackMyIndoorWorkout/issues/540#issuecomment-2586119809

However I also SDK API level limited some storage permissions. But the root cause was my faulty pub spec I think.
